roles/vault_pki/tasks/intermediate.yml

```diff
@@ -0,0 +1,82 @@
+---
+- name: "Ensure Vault Intermediate PKI backend exists"
+  hashivault_secret_engine:
+    url: "{{ vault_api_addr }}"
+    token: "{{ vault_token }}"
+    name: "{{ vault_pki_intermediate_ca_name }}"
+    description: "{{ vault_pki_intermediate_ca_name }} CA"
+    backend: "pki"
+
+- name: "Generate Intermediate CA cert, key and sign CSR"
+  block:
+    - name: "Generate Vault Intermediate CA cert and key"
+      hashivault_pki_ca:
+        url: "{{ vault_api_addr }}"
+        token: "{{ vault_token }}"
+        mount_point: "{{ vault_pki_intermediate_ca_name }}"
+        type: "{% if vault_pki_intermediate_export | bool %}exported{% else %}internal{% endif %}"
+        common_name: "{{ vault_pki_intermediate_ca_common_name }}"
+        kind: "intermediate"
+        config:
+          key_bits: 4096
```

Comment:
Let's get rid of those hardcoded values and put those as defaults, but overridable.